litestar/handlers/base.py

```diff
@@ -526,7 +527,7 @@ def _validate_dependency_is_unique(dependencies: dict[str, Provide], key: str, p
                     f"If you wish to override a provider, it must have the same key."
                 )
 
-    def on_registration(self, app: Litestar) -> None:
+    def on_registration(self, app: Litestar, route: BaseRoute) -> None:
```

Comment:
Not too psyched about having to pass in the route here, but then again we also pass it to the `ReceiveRoutePlugin` :woman_shrugging:

Reply:
Why?

`ReceiveRoutePlugin` was intentionally left undocumented. It was added to support the openapi generation refactor a while back, but the fact that is there doesn't bind us to iterating over the routes in this `Litestar.register()`.

Reply:
Hmm. We only need this information for the `HTTPRoute` to create the kwarg models for the different routes. In theory we could pass the path params, as that's the only thing that changes, but I thought that would be an even more leaky abstraction.

Since this PR is more of an intermediary step with the configuration changes, and the end goal is to get rid of the need for the `on_registration` callback entirely, I felt like this was an ok trade off to make for now as it's not *that* bad if we leave it in in the end. wdyt?

Reply:
Sounds good.